compiler/rustc_privacy/src/lib.rs

```diff
@@ -156,9 +156,10 @@ where
                 let leaf = leaf.subst(tcx, ct.substs);
                 self.visit_const(leaf)
             }
-            ACNode::Binop(..) | ACNode::UnaryOp(..) | ACNode::FunctionCall(_, _) => {
-                ControlFlow::CONTINUE
-            }
+            ACNode::Binop(..)
+            | ACNode::UnaryOp(..)
+            | ACNode::FunctionCall(_, _)
+            | ACNode::Cast(_, _, _) => ControlFlow::CONTINUE,
```

Comment:
I'm not sure if I should be `visit_ty`'ing on the type of the cast

Reply:
yea, you should. I don't think it is a problem right now, but this may permit unsizing casts in the future